clients/lsp-csharp.el

```diff
@@ -60,189 +60,53 @@ Usually this is to be set in your .dir-locals.el on the project root directory."
-                   target-version new-server-bin)))))))
-
-(defun lsp-csharp--get-or-install-server ()
-  "Resolves path to server binary installed, otherwise, if not found
-will ask the user if we can download and install it.
-Returns location of script or a binary to use to start the server."
-  (let ((installed-bin (lsp-csharp--server-bin (lsp-csharp--latest-installed-version))))
-    (if (and installed-bin (file-exists-p installed-bin))
-        installed-bin
-      (lsp-csharp--install-server nil t)
-      (let ((installed-bin (lsp-csharp--server-bin (lsp-csharp--latest-installed-version))))
-        (unless installed-bin (error "Server binary is required for LSP C# to work."))
-        installed-bin))))
-
-(defun lsp-csharp--download (url filename)
-  "Downloads file from URL as FILENAME. Will not do anything should
-the file exist already."
-  (unless (f-exists-p filename)
-    (message (format "lsp-csharp: downloading from \"%s\"..." url))
-    (let ((gnutls-algorithm-priority
-           (if (and (not gnutls-algorithm-priority)
-                    (boundp 'libgnutls-version)
-                    (>= libgnutls-version 30603)
-                    (version<= emacs-version "26.2"))
-               "NORMAL:-VERS-TLS1.3"
-             gnutls-algorithm-priority)))
-      (url-copy-file url filename nil))))
-
-(defun lsp-csharp--extract (filename target-dir)
-  "Extracts FILENAME which is a downloaded omnisharp-roslyn server
-tarball or a zip file (based on a current platform) to TARGET-DIR."
-  (cond
-   ((eq system-type 'windows-nt)
-    ;; on windows, we attempt to use powershell v5+, available on Windows 10+
-    (let ((powershell-version (substring
-                               (shell-command-to-string "powershell -command \"(Get-Host).Version.Major\"")
-                               0 -1)))
-      (if (>= (string-to-number powershell-version) 5)
-          (call-process "powershell"
-                        nil
-                        nil
-                        nil
-                        "-command"
-                        (concat "add-type -assembly system.io.compression.filesystem;"
-                                "[io.compression.zipfile]::ExtractToDirectory(\"" filename "\", \"" target-dir "\")"))
-
-        (message (concat "lsp-csharp: for automatic server installation procedure"
-                         " to work on Windows you need to have powershell v5+ installed")))))
-
-   ((or (eq system-type 'gnu/linux)
-        (eq system-type 'darwin))
-    (call-process "tar" nil nil t "xf" filename "-C" target-dir))
-
-   (t (error "lsp-csharp cannot extract \"%s\" on platform %s (yet)" filename system-type))))
+(defcustom lsp-csharp-omnisharp-roslyn-download-url
+  (format "https://github.com/omnisharp/omnisharp-roslyn/releases/latest/download/%s.zip"
+          (pcase system-type
+            ('gnu/linux "omnisharp-linux-x64")
+            ('darwin "omnisharp-osx")
+            ('windows-nt "omnisharp-win-x64")))
```

Comment:
Don't we need to care about this anymore?
> On Windows we're trying to avoid a crash starting 64bit .NET PE binaries in
Emacs by using x86 version of `omnisharp-roslyn` on older (<= 26.4) versions
of Emacs.

Reply:
Good catch! I have ammended the commit to restore checks.